storage/remote/azuread/README.md

```diff
@@ -0,0 +1,5 @@
+azuread provides an http.RoundTripper that attaches an Azure AD accessToken
+to remote write requests.
+
+This module is considered internal to Prometheus, without any stability
+guarantees for external usage.
```

Comment:
nit: should we use some formatting if this is a markdown file?

Reply:
Added some basic formatting